client/components/TestQueueCompletionStatusListItem/PreviouslyAutomatedTestCompletionStatus/index.js

```diff
@@ -0,0 +1,58 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { useQuery } from '@apollo/client';
+import { TEST_PLAN_RUN_ASSERTION_RESULTS_QUERY } from '../queries';
+import { isAssertionValidated } from '../../../utils/automation';
+
+const PreviouslyAutomatedTestCompletionStatus = ({
+    runnableTestsLength,
+    testPlanRunId,
+    id
+}) => {
+    const { data: testPlanRunAssertionsQueryResult } = useQuery(
+        TEST_PLAN_RUN_ASSERTION_RESULTS_QUERY,
+        {
+            variables: {
+                testPlanRunId: testPlanRunId
+            },
+            fetchPolicy: 'cache-and-network'
+        }
+    );
+
+    const numValidatedTests = useMemo(() => {
+        if (!testPlanRunAssertionsQueryResult) {
+            return 0;
+        }
+
+        const { testPlanRun } = testPlanRunAssertionsQueryResult;
+
+        if (!testPlanRun) {
+            return 0;
+        }
+
+        const { testResults } = testPlanRun;
+
+        return testResults?.reduce((acc, testResult) => {
```

Comment:
This component generally handles unavailable data by rendering a zero, except for this final condition where it would render the string "undefined". For consistency, how about:

```suggestion
        const { testResults = [] } = testPlanRun;

        return testResults.reduce((acc, testResult) => {
```

Reply:
Nice and clean. I sometimes forget that destructured properties can have default values.